src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+                    self._state = GoToObjectTaskState.waiting
+                else:
+                    self._state = GoToObjectTaskState.going
+                return (TaskRunning(), NopCommand())
+
+            elif self._state == GoToObjectTaskState.going:
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+                elif not (self._z_holder.check_z_error(
+                    self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Z error is too high'),)
+                elif not self._check_roomba_in_sight():
+                    return (TaskAborted(msg='The provided roomba is not in sight of quad'),)
+
+                try:
+                    roomba_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        self._roomba_id,
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('ObjectGoToTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during go_to roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                """ This is for tracking
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not found or not close enough to the quad'),)
+                """
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+                roomba_velocity = math.sqrt(roomba_x_velocity**2 + roomba_y_velocity**2)
+
+                x_overshoot = roomba_x_velocity/roomba_velocity * self._overshoot
+                y_overshoot = roomba_y_velocity/roomba_velocity * self._overshoot
+
+                # p-controller
+                x_vel_target = ((self._roomba_point.point.x + x_overshoot)
```

Comment:
The idea for this task is that it would use the `TranslateStopPlanner` class in order to go across the arena. So what I envision is you would create a `TranslateStopPlanner` every iteration of `get_desired_command` and pass in the current position of the roomba in the map frame, possibly with some overshoot that is dependent on the direction the roomba is moving and how far away from it we are, if that is known (which it won't always be known, so maybe we keep that out).

Reply:
Do you really wanted to be creating and destroying objects at 10Hz like that if you don't have to?

Reply:
@amiller27 and I talked about it before and we didn't see an issue with it.

Reply:
The translate stop planner will need to be refactored to support this. It loads a bunch of parameters from the param server and has a subscriber (and thus also deal with locks). It wasn't designed to be created and destroyed rapidly.

Reply:
Or would it be easier to refactor the `TranslateStopPlanner` to support new set points?

Reply:
And replanning

Reply:
IMO all the parameter loading and the subscriber need to be taken out if you want to create and destroy objects of the TranslateStopPlanner rapidly.

If that is done the translate stop planner devolves into essentially a method that you have to call a constructor on before it runs. All its parameters must be loaded in the class that uses it and they have to be passed in each time you create the object.

I agree that its more beautiful the way you've proposed. But in my judgement this crosses the line of practical vs nice code.

@amiller27 What do you think?

Reply:
I think the parameter loading is fine, but the way the whole `TranslateStopPlanner` works is probably an issue because a lot of it doesn't make sense if you only have one message on that topic that the subscriber is listening to

Reply:
I'm sorry I didn't understand Andrews first comment. Yes it would be easier to refactor the TranslateStopPlanner so that you can give it new setpoints. Creating and destroying it rapidly doesn't work with its current architecture and won't without significant functionality changes.

I do not understand why parameter loading and reloading over and over again is considered ok its incredibly wasteful.

What do you mean be not making sense if it only has one message? @amiller27 I think you're pointing out another reason why it wasn't designed to be created and destroyed rapidly.

I you guys want something that the translate stop planner is not which is a basically a planner math library.

Reply:
Aaron meant that it listens to the drone's current odometry, so for 1/10th of a second it will have old data, until the next one is created with an updated odometry from the task. So I think a refactor to accept new points is in order.

Reply:
How can it not operate on old data if its only ran at 10Hz?

Reply:
It listens the drone's odometry to determine where it is so it can plan accelerations accordingly. I think you want to rip that part out and just abstract out the trapezoidal planner part.

Reply:
Lots of things are wasteful.  The `TranslateStopPlanner` et. al. are limited by the rate of data coming into them, not the CPU.  I'm refining the meaning of "wasn't designed to be created and destroyed rapidly" by pointing out that the primary reason that's the case is because it doesn't give full functionality from the first message.

Reply:
I'm saying it WILL operate on old data if you just pass in the odometry every time, which isn't good at all.

Reply:
I'm claiming the parameter loading has the potential to be severely update rate limiting wasteful. We don't know yet so I believe tests should be done before jumping into this as a software engineering design pattern. It would be very nice if parameters can be loaded willy nilly all the time.

I see what you mean by requiring two odometery messages to enable full functionality. The reason that exists is because that planners algorithim depends on tracking state between each velocity request. Specifically it needs the last velocity request it made. If you use the odometry message each time you might never go anywhere because the controllers aren't perfect.  If you want to change that you need a new type of planner or you need to pass it in like its a math library.

Or you can just support changing the target position, we will need to do testing to make sure the algorithm handles that well.